docs/concepts/ksql-architecture.rst

```diff
@@ -0,0 +1,368 @@
+.. _ksql-architecture:
+
+KSQL Architecture
+#################
+
+The architecture of KSQL enables you to build real-time streaming applications
+from Kafka topics by using only SQL statements and queries. KSQL is built on
+Kafka Streams, so a KSQL application communicates with a Kafka cluster like any
+other Kafka Streams application.
+
+KSQL Components
+***************
+
+KSQL has four main components:
+
+* KSQL engine – processes KSQL statements and queries 
+* REST interface – enables client access to the engine
+* KSQL CLI – console that provides a command-line interface (CLI) to the engine
+* KSQL UI – enables developing KSQL applications in |c3|
+
+KSQL Server comprises the KSQL engine and the REST API. KSQL Server instances
+communicate with the Kafka cluster, and you can scale them as necessary.
+
+.. image:: ../img/ksql-architecture-and-components.png
+   :alt: Diagram showing architecture of KSQL
+   :align: center
+
+KSQL Engine
+    The KSQL engine executes KSQL statements and queries. You define your
+    application logic by writing KSQL statements, and the engine builds and
+    runs the application on available KSQL servers. Each KSQL server instance
+    runs a KSQL engine. Under the hood, the engine parses your KSQL statements
+    and builds corresponding Kafka Streams topologies.
+    
+    The KSQL engine is implemented in the
+    `KsqlEngine.java <https://github.com/confluentinc/ksql/blob/master/ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java>`__
+    class.
+
+KSQL CLI
+    The KSQL CLI provides a console with a command-line interface for the KSQL
+    engine. Use the KSQL CLI to interact with KSQL Server instances and develop
+    your streaming applications. The KSQL CLI is designed to be familiar to
+    users of MySQL, Postgres, and similar applications.
+
+    The KSQL CLI is implemented in the
+    `io.confluent.ksql.cli <https://github.com/confluentinc/ksql/tree/master/ksql-cli/src/main/java/io/confluent/ksql/cli>`__
+    package.
+
+REST Interface
+    The REST server interface enables communicating with the KSQL engine from
+    the CLI, |c3|, or from any other REST client. For more information, see
+    :ref:`ksql-rest-api`.
+    
+    The KSQL REST server is implemented in the
+    `KsqlRestApplication.java <https://github.com/confluentinc/ksql/blob/master/ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java>`__
+    class.
+
+When you deploy your KSQL application, it runs on KSQL Server instances that
+are independent of one another, are fault-tolerant, and can scale elastically
+with load. For more information, see :ref:`ksql-deployment-modes`.
+
+.. image:: ../img/ksql-server-scale-out.gif

```

Comment:
This animated image seems to load a bit slow. Does it need to be animated? 

![image](https://user-images.githubusercontent.com/11722533/50112698-c6c5a380-01f4-11e9-9fd4-b0614d7e81c6.png)

Reply:
I agree with Joel. They loaded slowly for me, and it's a bit much to absorb as they scroll by. Static images would be better IMHO.